src/cpp/src/visual_language/cdpruner/cdpruner.cpp

```diff
@@ -0,0 +1,627 @@
+// Copyright (C) 2023-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "cdpruner.hpp"
+
+#include <algorithm>
+#include <chrono>
+#include <cstring>
+#include <future>
+#include <iomanip>
+#include <iostream>
+#include <stdexcept>
+#include <thread>
+
+#include "openvino/openvino.hpp"
+#include "utils.hpp"
+
+namespace ov::genai::cdpruner {
+
+namespace {
+
+bool should_log_cdpruner(bool silent, bool debug_mode) {
+    if (ov::genai::utils::env_setup_for_print_debug_info()) {
+        return true;
+    }
+    if (debug_mode) {
+        return true;
+    }
+    return !silent;
+}
+
+}  // namespace
+
+CDPruner::CDPruner(const Config& config)
+    : m_config(config),
+      m_relevance_calc(config),
+      m_kernel_builder(config),
+      m_dpp_selector(config) {
+    // Load config from env
+    m_config.update_from_env();
+    // Validate configuration
+    validate_config(config);
+}
+
+std::vector<std::vector<size_t>> CDPruner::select_tokens(const ov::Tensor& visual_features,
+                                                         const ov::Tensor& text_features,
+                                                         bool silent) {
+    // Input validation
```

Comment:
This comment doesn't seem to express  any real meaning. please remove or update.